pkg/winperfcounters/watcher.go

```diff
@@ -151,24 +151,42 @@ func ExpandWildCardPath(counterPath string) ([]string, error) {
 	return win_perf_counters.ExpandWildCardPath(counterPath)
 }
 
-func removeTotalIfMultipleValues(vals []CounterValue) []CounterValue {
+// cleanupScrapedValues handles instance name collisions and standardizes names.
+// It cleans up the list in-place to avoid unnecessary copies.
+func cleanupScrapedValues(vals []CounterValue) []CounterValue {
 	if len(vals) == 0 {
 		return vals
 	}
 
-	if len(vals) == 1 {
-		// if there is only one item & the instance name is "_Total", clear the instance name
-		if vals[0].InstanceName == totalInstanceName {
-			vals[0].InstanceName = ""
-		}
+	// If there is only one "_Total" instance, clear the instance name.
+	if len(vals) == 1 && vals[0].InstanceName == totalInstanceName {
+		vals[0].InstanceName = ""
 		return vals
 	}
 
-	// if there is more than one item, remove an item that has the instance name "_Total"
-	for i, val := range vals {
-		if val.InstanceName == totalInstanceName {
-			return removeItemAt(vals, i)
+	occurrences := map[string]int{}
+	totalIndex := -1
+
+	for i := range vals {
+		instanceName := vals[i].InstanceName
+
+		if instanceName == totalInstanceName {
+			// Remember if a "_Total" instance was present.
+			totalIndex = i
 		}
+
+		if n, ok := occurrences[instanceName]; ok {
+			// Append indices to duplicate instance names.
+			occurrences[instanceName]++
+			vals[i].InstanceName = fmt.Sprintf("%s_%d", instanceName, n)
+		} else {
+			occurrences[instanceName] = 1
+		}
+	}
+
+	// Remove the "_Total" instance, as it can be computed with a sum aggregation.
```

Comment:
I think the reasoning for this is that if `_Total` is exported, any aggregation operation will likely double count (e.g. `sum()`) unless `_Total` is explicitly excluded. Also if you are scraping `*` you are likely to want to run aggregations.

I don't have a preference either way, but that's my guess as to why the code was written this way.

Reply:
Yes, this makes sense.

Reply:
That's a good point. I can play around with it a bit in the context of #29054. How would you like to proceed for this PR? I don't think it's directly related to the fix I'm making here, but it's definitely something that should be figured out.

Reply:
Yes, I think it is fine to proceed with the current PR. At this moment I'm leaning more in the direction of documenting the current behavior instead of doing any code change for #29054, but, before doing that I want to test a few things.